scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -1,93 +1,359 @@
+        # get rid of reset the index so that we don't get out of bounds errors
+        return df.dropna(subset=['ruling_citation']).reset_index(drop=True)
+
+    def get_citations(self, citations_as_string):
+        """
+        extract for each bger all ruling citations
+        :param citations_as_string:         citations how they were found in text of bger
+        :return:                            dataframe with additional column 'ruling_citation'
+        """
+        cits = []
         try:
-            supreme_court_df = next(self.select(engine, lang,
-                                                columns=f"{origin_chamber}, {origin_date}, {origin_file_number}",
-                                                where="court = 'CH_BGer'",
-                                                order_by="origin_date",
-                                                chunksize=self.get_chunksize()))
-        except StopIteration:
-            raise ValueError("No supreme court rulings found")
-        supreme_court_df = supreme_court_df.dropna(subset=['origin_date', 'origin_chamber'])
-        origin_chambers = list(supreme_court_df.origin_chamber.unique())
-        self.logger.info(f"Found supreme court rulings with references to lower court rulings "
-                         f"from chambers {origin_chambers}")
-        return origin_chambers, supreme_court_df
+            citations = ast.literal_eval(citations_as_string)  # parse dict string to dict again
+            for citation in citations:
+                try:
+                    cit = citation['text']
+                    citation_type = citation['name']
+                    cit = ' '.join(cit.split())  # remove multiple whitespaces inside
+                    if citation_type == "ruling":
+                        cited_file = self.get_file_number(cit)
+                        cits.append(cited_file)
+                    elif citation_type == "law":
+                        pass
+                    else:
+                        raise ValueError("type of citation must be 'rulings' or 'law")
+                except ValueError as ve:
+                    self.logger.info("citation has invalid syntax")
+                    self.logger.info(citation)
+                    continue
+        except ValueError as ve:
+            self.logger.info("citations could not be extracted to dict")
+            self.logger.info(citations_as_string)
+        if cits:  # only return something if we actually have citations
+            return cits
+
+    def get_file_number(self, citation):
+        """
+        find for each citation string the matching citation from the start of bge (first page)
+        :param citation:         citation as string as found in text
+        :return:                 RulingCitation always in German
+        """
+        # handle citation always in German
+        found_citation = RulingCitation(citation, 'de')
```

Comment:
Are there no citations in other languages?

Reply:
I want to check if the found citation is in my list of available bge. This list only contains file_numbers starting with BGE = German (I don't know why file_numbers are in German for all BGE). One could leave the starting string completely away since it doesn't contain any information. Citations about the same BGE can be made in different languages and would end with different starting strings, even though they cite the same BGE.

Reply:
Ok, I see. Yes maybe this would make more sense. In that case you can adapt the ruling_citation class to reflect this change.